examples/esp32c3/examples/sw_and_hw.rs

```diff
@@ -5,32 +5,32 @@
 mod app {
     use esp_backtrace as _;
     use esp_hal::{
-        gpio::{Event, Gpio9, Input, PullDown, IO},
+        gpio::{Event, GpioPin, Input, Io, Pull},
         peripherals::Peripherals,
-        prelude::*,
     };
     use esp_println::println;
+
     #[shared]
     struct Shared {}
 
     #[local]
     struct Local {
-        button: Gpio9<Input<PullDown>>,
+        button: Input<'static, GpioPin<9>>,
     }
 
     // do nothing in init
     #[init]
     fn init(_: init::Context) -> (Shared, Local) {
         println!("init");
         let peripherals = Peripherals::take();
-        let io = IO::new(peripherals.GPIO, peripherals.IO_MUX);
-        let mut button = io.pins.gpio9.into_pull_down_input();
+        let io = Io::new_no_bind_interrupt(peripherals.GPIO, peripherals.IO_MUX);
```

Comment:
I will note that this is a bit of a rough edge for which we don't have a good solution yet. Essentially the issue is that `Io::new()` binds a handler to the `GPIO` interrupt for async functionality, so if we want to use this interrupt for other purposes then we need to _not_ bind the interrupt in the constructor.

This is potentially a bit of a footgun, so I can add a comment here if necessary.

Reply:
There is a tracking issue for this in the `esp-hal` repo, FYI:  
https://github.com/esp-rs/esp-hal/issues/2009